File: pedro_vitor_problema2.py
import math

def calcComprimento(a, x0, x1, e, N):

    f0 = 0
    f1 = 0
    d1 = 0
    d2 = 20
    n = 2
    
    while abs(d1 - d2) > e:

        d2 = d1
        h = (x0-x1)/n

        for i in range(N):

            x1 = x0 + h

            for j in range(N):
                f0 += a[j]*pow(x0,j)
                f1 += a[j]*pow(x1,j)
            
            d1 += math.sqrt(pow(f1-f0,2) + pow(x1-x0,2))
            x0 = x1
            
        n*=2
        
    return d1


N = int(input("Digite o grau de seu polinomio:"))
coef = list()

for i in range(N+1):
    a = float(input("Digite o coeficiente"))
    coef.append(float(a))

x0 = float(input("Digite o limite inferior do polinomio:"))
x1 = float(input("Digite o limite superior do polinomio:"))
precisao = float(input("Digite a precisao desejada:"))

comprimento = calcComprimento(coef, x0, x1, precisao, N+1)

print("Comprimento vale:")
print(comprimento)

File: pedro_vitor_problema2.c
#include <stdio.h>
#include <stdlib.h>
#include <math.h>


float calculaComp( float *, float , float , float , int);

int main ()
{
    int i, N;
    float *coef;
    float x0, x1, precisao;
    float comprimento;

    printf("Digite o grau de seu polinomio:\n ");
    scanf("%d" , &N);

    coef = (float *) malloc((N+1)*sizeof(float));

    for( i = 0; i < (N+1); i++)
    {
        printf("Digite coeficiente de grau %d de seu polinomio:\n ", N-i);
        scanf("%f" , coef + N - i);
    }

    printf("Digite o limite inferior de seu polinomio:\n ", N-i);
    scanf("%f" , &x0);

    printf("Digite o limite superior de seu polinomio:\n ", N-i);
    scanf("%f" , &x1);

    printf("Digite o valor da precisao desejada:\n ", N-i);
    scanf("%f" , &precisao);

    comprimento = calculaComp(coef, x0, x1, precisao, N+1);

    printf("\nComprimento vale: %f\n", comprimento);

    free(coef);

    system("pause");

    return 0;
}


float calculaComp( float *a, float x0, float x1, float e, int N)
{
    float f0 = 0;
    float f1 = 0;
    float d1 = 0, d2 = 20;
    int i, j;
    int n = 2;
    float h;

    // x1 = x0 + h;
    
    while( fabs(d1-d2) > e )
    {
        
        d2 = d1;
        h = (x0-x1)/n;

        for( i = 0; i < n; i++)
        {
            x1 = x0 + h;

            for( j = 0; i < N; i++)
            {
                f0 += a[i]*pow(x0,i);
                f1 += a[i]*pow(x1,i);
            }

            d1 += sqrt( pow(f1-f0,2) + pow(x1-x0,2));

            x0 = x1;
        
        }

        n *= 2;

    }

    return d1;
}
